agent/consul/leader_connect_ca.go

```diff
@@ -1139,81 +1136,113 @@ func (c *CAManager) runRenewIntermediate(ctx context.Context) error {
 // RenewIntermediate checks the intermediate cert for
 // expiration. If more than half the time a cert is valid has passed,
 // it will try to renew it.
-func (c *CAManager) RenewIntermediate(ctx context.Context, isPrimary bool) error {
+func (c *CAManager) RenewIntermediate(ctx context.Context) error {
+	_, err := c.renewIntermediate(ctx, false, false)
+	return err
+}
+
+func (c *CAManager) renewIntermediateNow(ctx context.Context) error {
+	renewed, err := c.renewIntermediate(ctx, true, true)
+	if err != nil {
+		return err
+	}
+	if !renewed {
+		return errors.New("expected to force renew and did not")
+	}
```

Comment:
It feels awkward to add a `bool` return variable for this case which currently isn't reachable. Is there a need to be defensive about it? I can't see `forceRenewAheadOfSchedule` being misused for anything else other than a guard for the lifetime check.

Reply:
That's fair. I guess the test would fail if it didn't renew. I'll remove it.